bundles/com.espressif.idf.core/src/com/espressif/idf/core/build/ErrorMarkerListener.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright 2024 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.build;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IMarkerDelta;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Platform;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.IDFCorePreferenceConstants;
+import com.espressif.idf.core.IDFEnvironmentVariables;
+import com.espressif.idf.core.logging.Logger;
+
+/**
+ * Marker listener that can listen to added markers in the problems view and start a job to remove them from the derived
+ * files in ide folder for esp_idf components
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class ErrorMarkerListener implements IResourceChangeListener
+{
+	private static String IDF_PATH;
+
+	public ErrorMarkerListener()
+	{
+		IDF_PATH = new IDFEnvironmentVariables().getEnvValue(IDFEnvironmentVariables.IDF_PATH);
```

Comment:
Do we need to initialize the IDF_PATH field in the constructor? As far as I understand, we assign this listener when starting the IDE. In this case, the IDF_PATH will be null on the first startup, when the environment variables have not yet been initialized. Also, this variable will not be changed after the tools are installed.

Reply:
Fixed